### PR TITLE
Split the qesap regression test module

### PR DESCRIPTION
**Pull request by mpagot**
Split the module about testing the deployment in 4 modules each one to focus on a different test area. Introduce a new test step to trigger a Crash in a node.

- Related ticket: https://jira.suse.com/browse/TEAM-9654

# Verification run:

### sle-15-SP6-Qesap-Azure-Byos-x86_64-BuildLATEST_AZURE_SLE15_6_BYOS-qesap_azure_saptune_test
- http://openqaworker15.qa.suse.cz/tests/296642 :green_circle:  there are now 4 test module named `test_*`. Mirror test module correctly detect the setting and skip most of the stage that are not applicable.

## sle-15-SP5-Qesap-Azure-Byos-x86_64-BuildLATEST_AZURE_SLE15_5_BYOS-qesap_azure_ibsmirror_peering_test
 - http://openqaworker15.qa.suse.cz/tests/296643 :green_circle:  test_mirror is running all the Azure related checks.

###  - sle-15-SP6-Qesap-Aws-Byos-x86_64-BuildLATEST_AWS_SLE15_6_BYOS-qesap_aws_ibsmirror_peering_test
- http://openqaworker15.qa.suse.cz/tests/296867 :red_circle: fails in `test_mirror` but it systematically fails in existing code too http://openqaworker15.qa.suse.cz/tests/297022#next_previous
 - sle-15-SP6-Qesap-Aws-Byos-x86_64-BuildLATEST_AWS_SLE15_6_BYOS-qesap_aws_ibsmirror_peering_test@64bit -> http://openqaworker15.qa.suse.cz/tests/297127 after merging https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20171

### CODE REVIEW

**Review by lilyeyes — APPROVED** [review]
Generally  lgtm.

File: tests/sles4sap/qesapdeployment/test_cluster.pm [hunk 1]
```diff
@@ -53,56 +41,13 @@ sub run {
...
-    elsif ($provider_setting eq 'EC2') {
-        if (get_var("QESAPDEPLOY_IBSMIRROR_IP_RANGE")) {
-            my $deployment_name = qesap_calculate_deployment_name('qesapval');
-            my $vpc_id = qesap_aws_get_vpc_id(resource_group => $deployment_name);
-            my $ibs_mirror_target_ip = get_var('QESAPDEPLOY_IBSMIRROR_IP_RANGE');    # '10.254.254.240/28'
-            die 'Error in network peering setup.' if !qesap_aws_vnet_peering(target_ip => $ibs_mirror_target_ip, vpc_id => $vpc_id);
-            qesap_add_server_to_hosts(name => 'download.suse.de', ip => get_required_var("QESAPDEPLOY_IBSMIRROR_IP"));
-            die 'Error in network peering delete.' if !qesap_aws_delete_transit_gateway_vpc_attachment(name => $deployment_name . '*');
-        }
-    }
```
Comment (BillAnastasiadis) [inline tests/sles4sap/qesapdeployment/test_cluster.pm:75]: Why is all that nw peering code above removed? Just out of curiosity

Comment (BillAnastasiadis) [reply tests/sles4sap/qesapdeployment/test_cluster.pm:75]: ah, ok, you moved it to `test_mirror`. Cool

**Review by BillAnastasiadis — APPROVED** [review]
LGTM

**Review by Amrysliu — APPROVED** [review]
LGTM

**Pull request merged**